### ISSUE TITLE
Error is not reported on `Type[Protocol]`, which will raise `TypeError` at runtime

### ISSUE THREAD

**Issue by kennipj** [microsoft/pyright#2127]
No error is reported when annotating with `Type[Protocol]`, which results in a runtime error:
```TypeError: Plain <class 'typing.Protocol'> is not valid as type argument```

Sample code:
```py
from typing import Protocol, Type

def some_func(protocol: Type[Protocol]):
    # TypeError: Plain <class 'typing.Protocol'> is not valid as type argument
    pass

```

Mypy will report an error in this case: ``` error: Variable "typing.Protocol" is not valid as a type```, and I would expect pyright to do the same in this case.

Feels like a bug in the typing module, as it limits typing rather significantly when trying to extend functionality on protocol checking, however that is beside the point here.

Tested with:
pyright `1.1.158`
mypy `0.910`

**Comment by erictraut** [comment]
Thanks for the suggestion. This error case will be included in the next release.

**Comment by erictraut** [comment]
This is now addressed in pyright 1.1.159, which I just published. It will also be included in the next release of pylance.
